File: icure-sdk/src/commonMain/kotlin/com/icure/sdk/model/filter/ComplementFilter.kt
package com.icure.sdk.model.filter

import com.icure.sdk.model.base.Identifiable
import kotlin.String
import kotlinx.serialization.Serializable

@Serializable
public data class ComplementFilter<O : Identifiable<String>>(
  override val desc: String? = null,
  override val superSet: AbstractFilter<O>,
  override val subSet: AbstractFilter<O>,
) : AbstractFilter<O>, Filter.ComplementFilter<O>

File: python-wrapper/src/python/model/filter/ComplementFilter.py
import json
from model.filter.AbstractFilter import AbstractFilter, serialize_abstract_filter, deserialize_abstract_filter
from typing import Optional, Union, Dict
from dataclasses import dataclass

@dataclass
class ComplementFilter:
	super_set: 'AbstractFilter'
	sub_set: 'AbstractFilter'
	desc: Optional[str] = None

	def __serialize__(self) -> object:
		return {
			"desc": self.desc,
			"superSet": serialize_abstract_filter(self.super_set),
			"subSet": serialize_abstract_filter(self.sub_set),
		}

	@classmethod
	def _deserialize(cls, data: Union[str, Dict[str, object]]) -> 'ComplementFilter':
		deserialized_dict: dict[str, object]
		if isinstance(data, str):
			deserialized_dict = json.loads(data)
		else:
			deserialized_dict = data
		return cls(
			desc = deserialized_dict.get("desc"),
			super_set = deserialize_abstract_filter(deserialized_dict["superSet"]),
			sub_set = deserialize_abstract_filter(deserialized_dict["subSet"]),
		)

__all__ = ['ComplementFilter']